File: inutile/xx.ml
open Yojson.Basic

type data_item = {
  id : int;
  value : int;
}

let generate_data () =
  Random.self_init ();
  let data = Array.init 10 (fun i -> { id = i; value = Random.int 100 }) in
  let json_data = `List (Array.to_list (Array.map (fun item -> `Assoc [("id", `Int item.id); ("value", `Int item.value)]) data)) in
  let json_string = Yojson.Basic.pretty_to_string json_data in
  let oc = open_out "data.json" in
  output_string oc json_string;
  close_out oc;
  print_endline "Data generated and saved to data.json"

let () = generate_data ()

File: inutile/xx.ts
import * as fs from 'fs';

interface DataItem {
    id: number;
    value: number;
}

const data: DataItem[] = [];

for (let i = 0; i < 10; i++) {
    data.push({ id: i, value: Math.floor(Math.random() * 100) });
}

const jsonData = JSON.stringify(data, null, 4);
fs.writeFileSync('data.json', jsonData);

console.log('Data generated and saved to data.json');